integration/error-boundary-test.ts

```diff
@@ -636,3 +637,241 @@ test.describe("ErrorBoundary", () => {
+    console.error = () => {};
+    consoleErrors = [];
+    // Listen for all console events and handle errors
+    page.on("console", (msg) => {
+      if (msg.type() === "error") {
+        consoleErrors.push(msg.text());
+      }
+    });
+  });
+
+  test.afterEach(() => {
+    console.error = oldConsoleError;
+  });
+
+  test.describe("without JavaScript", () => {
+    test.use({ javaScriptEnabled: false });
+    runBoundaryTests();
+  });
+
+  test.describe("with JavaScript", () => {
+    test.use({ javaScriptEnabled: true });
+    runBoundaryTests();
+  });
+
+  function runBoundaryTests() {
+    test("Prevents useLoaderData in self ErrorBoundary", async ({
+      page,
+      javaScriptEnabled,
+    }) => {
+      let app = new PlaywrightFixture(appFixture, page);
+      await app.goto("/parent/child-with-boundary");
+
+      expect(await app.getHtml("#parent-data")).toEqual(
+        '<p id="parent-data">PARENT</p>'
+      );
+      expect(await app.getHtml("#child-data")).toEqual(
+        '<p id="child-data">CHILD</p>'
+      );
+      expect(consoleErrors).toEqual([]);
+
+      await app.clickSubmitButton("/parent/child-with-boundary");
+      await page.waitForSelector("#child-error");
+
+      expect(await app.getHtml("#child-error")).toEqual(
+        '<p id="child-error">Broken!</p>'
+      );
+      expect(await app.getHtml("#parent-data")).toEqual(
+        '<p id="parent-data">PARENT</p>'
+      );
+      expect(await app.getHtml("#child-data")).toEqual(
+        '<p id="child-data"></p>'
+      );
+
+      if (javaScriptEnabled) {
+        expect(consoleErrors).toEqual([
+          "Failed to load resource: the server responded with a status of 500 (Internal Server Error)",
+          "You cannot `useLoaderData` in an errorElement (routeId: routes/parent/child-with-boundary)",
+        ]);
+      } else {
+        expect(consoleErrors).toEqual([
+          "Failed to load resource: the server responded with a status of 500 (Internal Server Error)",
+          // We don't get the useLoaderData message in the client when JS is disabled
+        ]);
+      }
+    });
+
+    test("Prevents useLoaderData in bubbled ErrorBoundary", async ({
+      page,
+      javaScriptEnabled,
+    }) => {
+      let app = new PlaywrightFixture(appFixture, page);
+      await app.goto("/parent/child-without-boundary");
+
+      expect(await app.getHtml("#parent-data")).toEqual(
+        '<p id="parent-data">PARENT</p>'
+      );
+      expect(await app.getHtml("#child-data")).toEqual(
+        '<p id="child-data">CHILD</p>'
+      );
+      expect(consoleErrors).toEqual([]);
+
+      await app.clickSubmitButton("/parent/child-without-boundary");
+      await page.waitForSelector("#parent-error");
+
+      expect(await app.getHtml("#parent-error")).toEqual(
+        '<p id="parent-error">Broken!</p>'
+      );
+      expect(await app.getHtml("#parent-matches-data")).toEqual(
+        '<p id="parent-matches-data"></p>'
```

Comment:
FWIW I think this is a bug in TM today that is fixed in the new router.  When JS is enabled, TM doesn't clear out the prior parent `routeData` so we still get a now-stale `PARENT` here even though the loader was not re-run